dao-rest/src/main/java/com/nikolay/client/DepartmentRestDaoImpl.java

```diff
@@ -0,0 +1,109 @@
+package com.nikolay.client;
+
+import com.nikolay.client.exception.ServerDataAccessException;
+import com.nikolay.model.Department;
+import com.nikolay.service.DepartmentService;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * The type Department rest dao.
+ */
+@Component
+public class DepartmentRestDaoImpl implements DepartmentService {
+
+    /**
+     * The constant LOGGER.
+     */
+    public static final Logger LOGGER = LogManager.getLogger();
+
+    @Value("${app.url}")
+    private String url;
+
+    @Value("${point.departments}")
+    private String departmentsPoint;
```

Comment:
may be it is more suitable to use one url property instead several values which always are concatenated